lib/spark.js

```diff
@@ -169,7 +173,12 @@ export default class Spark {
         }
         console.error(err)
       }
-      await sleep(DELAY_BETWEEN_RETRIEVALS)
+      const duration = Date.now() - started
+      const baseDelay = APPROX_ROUND_LENGTH_IN_MS / this.#maxTasksPerNode
```

Comment:
what about instead having `this.#delayBetweenRetrievals = APPROX_ROUND_LENGTH_IN_MS / maxTasksPerNode` and storing that in `getRetrieval()`?

And then something like

```js
await sleep(Math.max(0, this.#delayBetweenRetrievals - duration))
```

Reply:
See https://github.com/filecoin-station/spark-api/pull/172#discussion_r1423969809

I designed the current solution with a longer-term vision in my mind.

If you feel strongly this is YAGNI, then I can rework this PR to apply your suggestion. Let me know!